src/configure/clients/ProvisioningServiceClient.ts

```diff
@@ -0,0 +1,46 @@
+import { ServiceClientCredentials, UrlBasedRequestPrepareOptions } from "ms-rest";
+import { ProvisioningConfiguration } from "../model/provisioningConfiguration";
+import { IProvisioningServiceClient } from "./IProvisioningServiceClient";
+import { RestClient } from "./restClient";
+
+export class ProvisioningServiceClient implements IProvisioningServiceClient {
+  private restClient: RestClient;
+  private baseUrl: string;
+  private defaultHeaders: { [propertyName: string]: string };
+  private readonly pipelineProvisioningJob = "PipelineProvisioningJob";
+  constructor( baseUrl: string, githubPAT: string, credentials: ServiceClientCredentials) {
+    this.restClient = new RestClient(credentials);
+    this.baseUrl = baseUrl;
+    this.defaultHeaders = {
+      "Content-Type": "application/json",
+      "X-GITHUB-TOKEN": githubPAT,
```

Comment:
pat token header is only required if baseUrl is PE service otherwise it will go as Authorization header.
Also api-version query version is needed for PE service.

Reply:
okay, then i will add Headers as part of constructor . For api-version , we have optional param queryParam createProvisioning/getProvisioning request which can be used to pass api-version

Reply:
api-version should be part of ProvisioningServiceClient.ts, consumer of this should not worry about the api version. You can store it in default query params based on baseUrl.

Reply:
Updated with changes